src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Headers.cs

```diff
@@ -97,6 +97,32 @@ public async Task SendAsync_SpecialCharacterHeader_Success()
             });
         }
 
+        [Fact]
+        public async Task SendAsync_EmptyResponseHeader_Success()
+        {
+            await LoopbackServerFactory.CreateClientAndServerAsync(async uri =>
+            {
+                using (var client = CreateHttpClient())
+                {
+                    var message = new HttpRequestMessage(HttpMethod.Get, uri);
+                    var response = await  client.SendAsync(message).ConfigureAwait(false);
+
+                    // HTTP/1.1 adds Connection: close and Date
```

Comment:
Can you expand on this comment to more fully explain why the numbers "3" and "5" below are the right numbers?

Reply:
3 is expected because we send 3 in the block bellow. I tried to comment on 5 as old loopback adds headers on its own and that i less obvious. For a moment, I thought about making them same but Adding "Connection: close" to Http2 does not make sense. 
If you have suggestion on better wording let me know.

Reply:
>3 is expected because we send 3 in the block bellow

Looking now I see that. It would be better if the code then were written like this:

```c#
Assert.Equal(UseHttp2LoopServer ? headers.Length : headers.Length + 2, response.Headers.Count());
```

Or something like that. That might be hard since you might have to move the `headers` definition out of the async lambda and define it in the body of the test method. For product code, that would make performance a little less. But since this is test code, I think readability and maintainability of this is more important.

Reply:
Updated. And I add coverage for TrailingHeaders as well.